src/app/modules/shared-components/components/textarea/textarea.component.ts

```diff
@@ -1,29 +1,27 @@
-import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { Component, Input, Output, EventEmitter, OnInit, OnChanges } from '@angular/core';
+import { FormGroup } from '@angular/forms';
 
 @Component({
   selector: 'alg-textarea',
   templateUrl: './textarea.component.html',
   styleUrls: [ './textarea.component.scss' ],
 })
-export class TextareaComponent implements OnInit {
+export class TextareaComponent implements OnInit, OnChanges {
+  @Input() inputName = ''; // name of the input in the parent form
+  @Input() parentForm?: FormGroup;
+
   @Input() icon = '';
   @Input() placeholder = '';
   @Input() disabled = false;
 
   @Output() textChange = new EventEmitter<string>();
 
-  value = '';
-
   constructor() {}
 
   ngOnInit(): void {}
 
-  setValue(text: string): void {
-    this.value = text;
-    this.onChange();
-  }
-
-  onChange(): void {
-    this.textChange.emit(this.value);
+  ngOnChanges(): void {
+    if (this.disabled) this.parentForm?.disable();
```

Comment:
I don't think the "disabled" property should be kept is the way to do it is to used the formgroup. At the end, it means that is the parent which ask his child to disable itself, which itself change a state of the parent...

Reply:
The way 'disabled' is done is different now.